.grit/patterns/js/react_hooks.grit

```diff
@@ -45,18 +45,15 @@ private pattern handle_one_statement($class_name, $statements, $states_statement
                 and {
                     $name <: js"constructor",
                     $body <: maybe contains bubble($constructor_statements) {
-                        or {
-                            lexical_declaration($declarations) as $decl where $declarations <:
-                                variable_declarator($name, $value) where {
-                                    $name <: not contains js"this.state",
-                                    $value <: not contains js"this.state",
-                                    $constructor_statements += $decl
+                        statement_block($statements) where {
+                            $statements <: some bubble($constructor_statements) $statement where {
+                                $constructor_statements <: not contains $statement,
```

Comment:
This condition prevents adding the same statement to the `$constructor_statements` list, but I'm unsure if this is the right way to handle this. 

Example of what this condition prevents from:

<table>
<tr>
<td>

```js
if (Match.random() < 2) {
  console.log('Win!');
}
```

</td>
<td>

`=>`

</td>
<td>

```js
if (Match.random() < 2) {
  console.log('Win!');
}

console.log('Win!');
```

</td>
</tr>
</table>

Reply:
You should be using `some` and iterating explicitly over the statements, not using `contains` which will (as you can see) recurse and cause major issues.